docs/CLIUsage.md

```diff
@@ -1,23 +1,120 @@
+            * [Install a Framework](#install-a-framework)
+               * [Install just the Framework without dependencies](#install-just-the-framework-without-dependencies)
+            * [Install a Framework with Dependencies](#install-a-framework-with-dependencies)
+            * [Install a Framework with InstanceName &amp; Parameters](#install-a-framework-with-instancename--parameters)
+            * [List Instances](#list-instances)
+            * [Get the Status of an Instance](#get-the-status-of-an-instance)
+            * [Get the History to PlanExecutions](#get-the-history-to-planexecutions)
+
+## Kubectl KUDO Plugin
+
+### Requirements
+
+- KUDO client automatically created via `code-generator`
+- GitHub set up
+  - GitHub OTP token in `$HOME/.git-credentials` OR
+  - GitHub Basic Auth via `GIT_USER` and `GIT_PASSWORD` environment variables exposed
 
-There are two CLIs you can use at the moment:
+### Install
+
+Install the plugin from your `$GOPATH/src/github.com/kudobuilder/kudo` root folder via:
+
+- `go install ./cmd/kubectl-kudo`
+
+### Commands
+
+|  Syntax | Description  | 
+|---|---|
+| `kubectl kudo install <name> [flags]`  |  Installs a Framework from the official [KUDO repo](https://github.com/kudobuilder/frameworks). |
+| `kubectl kudo list instances [flags]` | Show all available instances. | 
+| `kubectl kudo plan status [flags]` | View all available plans. |
+| `kubectl kudo plan history <name> [flags]` | View all available plans. |
+
+### Flags
+
+|  Name | Shorthand | Default | Usage  | 
+|---|---|---|---|
+| repo-version | | | A specific repo version on the official GitHub repo; defaults to the most recent |
+| all-dependencies |  | true | Installs all dependency frameworks (with frameworkversions) it can find from the dependency section in `frameworkversions`|
+| auto-approve | | false | Skip interactive approval when existing version found. To always install the latest framework/frameworkversion |  
+| githubcredential | | | The file path to github credential file; defaults to $HOME/.git-credentials |
+| instance | | Framework name | The instance name. (default to Framework name) |
+| namespace | | "default" | The namespace where the operator watches for changes |
+
+### Environment Variables
+
+Environment Variables override flags and are intended to give the user more customizable CLI options.
+
+|  Environment Variable | Description  | 
+|---|---|
+| `GIT_USER`  |  Set a GitHub user to connect via the GitHub API with |
+| `GIT_PASSWORD` | Set a GitHub password to connect via the GitHub API with | 
+
+### Examples
+
+#### Install a Framework
+
+There are two options installing a Framework. While `kubectl apply -f *.yaml` is encouraged when developing and
+having `framework.yaml` or `frameworkversion.yaml` locally present, it is recommended to use the official Frameworks 
+provided in our [kubebuilder/frameworks](https://github.com/kudobuilder/frameworks) repo. `Kudoctl` offers a convenient
```

Comment:
Okay so then that makes me think:
We call it `kubectl-kudo` to follow the Kubectl Plugin System in that it enables us to use it the way `kubectl kudo ...` but at the same time when you run `go install ./cmd/kubectl-kudo
` a `kubectl-kudo` binary will be copied to your `go/bin` path that you could run standalone. If we would follow the logic I just see that we then would need to call it `kubectl-kudo` instead of `kudoctl`. Or we would call it `kudo` plugin for `kubectl`. Wdyt?

Reply:
on that note: should we rename `/pkg/kudoctl` then into `/pkg/kudo` to keep it persistent?

Reply:
I don't think many people will install it via`go install ./cmd/kubectl-kudo`, since we should eventually have it wrapped up in a Makefile.

lets try and find some other plugins and see what they do for refernce.